common/src/main/java/com/wynntils/wc/objects/item/WynnGearStack.java

```diff
@@ -0,0 +1,418 @@
+
+    public WynnGearStack(ItemStack stack) {
+        super(stack);
+
+        // get item profile
+        if (WebManager.getItemsMap() == null || !WebManager.getItemsMap().containsKey(itemName)) return;
+        itemProfile = WebManager.getItemsMap().get(itemName);
+
+        // identification parsing & tooltip creation
+        identifications = new ArrayList<>();
+
+        List<Component> lore =
+                ComponentUtils.stripDuplicateBlank(super.getTooltipLines(null, TooltipFlag.Default.NORMAL));
+        lore.remove(0); // remove item name
+
+        List<Component> baseTooltip = new ArrayList<>();
+
+        int idAmount = 0; // only counting non-fixed IDs known by the API
+        float percentTotal = 0;
+        boolean hasNew = false;
+
+        boolean hasIds = false;
+        boolean endOfIDs = false;
+        for (int i = 0; i < lore.size(); i++) {
+            Component loreLine = lore.get(i);
+            String unformattedLoreLine = WynnUtils.normalizeBadString(loreLine.getString());
+
+            if (unformattedLoreLine.equals("Set Bonus:")) {
+                endOfIDs = true;
+            }
+
+            if (endOfIDs) continue;
+
+            ItemIdentificationContainer idContainer = ItemIdentificationContainer.fromLore(loreLine, itemProfile);
+            if (idContainer == null) { // not an ID line
+                baseTooltip.add(loreLine);
+                continue;
+            }
+
+            // if we've reached this point, we have an id
+            if (!hasIds) {
+                hasIds = true;
+                baseTooltip.add(ID_PLACEHOLDER);
+                baseTooltip.add(new TextComponent(""));
+            }
+
+            identifications.add(idContainer);
+
+            if (!idContainer.isNew() && !idContainer.isFixed()) {
+                idAmount++;
+                percentTotal += idContainer.getPercent();
+            }
+
+            if (idContainer.isNew()) {
+                hasNew = true;
+            }
+        }
+
+        if (!identifications.isEmpty()) {
+            baseTooltip = ComponentUtils.stripDuplicateBlank(baseTooltip);
+        }
+
+        constructTooltips(baseTooltip);
+
+        // overall percent & name
+        overallPercentage = -1f;
+
+        String originalName = WynnUtils.normalizeBadString(ComponentUtils.getUnformatted(getHoverName()));
+        MutableComponent name = new TextComponent(originalName);
+
+        if (hasNew) {
+            name.append(new TextComponent(" [NEW]").withStyle(ChatFormatting.GOLD));
+        } else if (idAmount > 0) {
+            overallPercentage = percentTotal / idAmount;
+
+            // check for perfect/0% items
+            isPerfect = overallPercentage >= 100d;
+            isDefective = overallPercentage == 0;
+
+            name.append(ItemStatInfoFeature.getPercentageTextComponent(overallPercentage));
+        }
+
+        customName = name;
+    }
+
+    public WynnGearStack(ItemProfile profile) {
```

Comment:
When is this constructor used?

Reply:
this one is intended for future functionality with item guides and the like. [here's an example item created from it](https://media.discordapp.net/attachments/942564848330498048/973847788779601980/unknown.png)

Reply:
i would consider the item guide one of the most useful and universally liked features of original wynntils, so i expect it to get re-implemented eventually. it's hard to envision a way of creating it without using some type of itemstack.

`constructBaseTooltip()` is essentially required for several other important features too, such as a player gear viewer or chat item representations